lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystem.java

```diff
@@ -202,4 +211,34 @@ private static void validateS3Location(Location location)
     {
         new S3Location(location);
     }
+
+    private boolean shouldReadObject(S3Object object)
+    {
+        return switch (context.s3ObjectStorageClassFilter()) {
+            case READ_ALL -> true;
+            case SKIP_ALL_GLACIER -> !isGlacierObject(object);
+            case READ_RESTORED_GLACIER_OBJECTS -> !isGlacierObject(object) || isCompletedRestoredObject(object);
```

Comment:
That will restore ALL non Glacier objects and RESTORED objects ? Should we adjust name ?
Maybe 
READ_ALL
READ_ALL_NON_GLACIER
READ_ALL_NON_GLACIER_AND_RESTORED

Reply:
Agreed this is more clear will update